storage/src/main/java/com/brandwatch/ahmedanalytics/storage/Config.java

```diff
@@ -0,0 +1,32 @@
+package com.brandwatch.ahmedanalytics.storage;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+import java.util.Properties;
+
+@Configuration
+@EnableScheduling
+public class Config {
+
+    @Bean
+    public KafkaConsumer<String, String> mentionConsumer() {
+        Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "kafka:9092");
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, "storage-consumer");
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+        props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+
+        return new KafkaConsumer<>(props);
+    }
+
+    @Bean(initMethod = "run", destroyMethod = "shutdown")
```

Comment:
initMethod should be `start`, not `run`. If run is called, spring initialisation will just call the method, but not start a new thread. The main thread of Spring will get stuck in infinite loop of kafka consumer and may never finish initialisation of the context.

You can read javadocs for Thread class and see the difference in methods.

Reply:
I know the difference between them since I fell in this trap before but in python as I can remeber 😅. I changed it